app/src/main/java/org/mozilla/focus/telemetry/GleanMetricsService.kt

```diff
@@ -0,0 +1,104 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.telemetry
+
+import android.content.Context
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers.IO
+import kotlinx.coroutines.MainScope
+import kotlinx.coroutines.async
+import kotlinx.coroutines.launch
+import mozilla.components.browser.state.search.SearchEngine
+import mozilla.components.browser.state.state.selectedOrDefaultSearchEngine
+import mozilla.components.feature.search.ext.waitForSelectedOrDefaultSearchEngine
+import mozilla.components.service.glean.net.ConceptFetchHttpUploader
+import mozilla.telemetry.glean.Glean
+import mozilla.telemetry.glean.config.Configuration
+import org.mozilla.focus.BuildConfig
+import org.mozilla.focus.Components
+import org.mozilla.focus.GleanMetrics.GleanBuildInfo
+import org.mozilla.focus.GleanMetrics.Browser
+import org.mozilla.focus.GleanMetrics.LegacyIds
+import org.mozilla.focus.GleanMetrics.Pings
+import org.mozilla.focus.ext.components
+import org.mozilla.focus.telemetry.TelemetryWrapper.isTelemetryEnabled
+import org.mozilla.focus.utils.Settings
+import java.util.UUID
+
+/**
+ * Glean telemetry service.
+ *
+ * To track events, use Glean's generated bindings directly.
+ */
+class GleanMetricsService(context: Context) : MetricsService {
+
+    private val activationPing = ActivationPing(context)
+
+    /**
+     * Initialization flag to
+     */
+    private var initialized = false
+
+    override fun initialize(context: Context) {
+        if (initialized) {
+            return
+        }
+        initialized = true
```

Comment:
Is this flag needed? It seems to only get called from `FocusApplication.onCreate()`.

Reply:
Ah good point, this isn't needed any more since we're using Glean directly and do not need to consider about multiple calls to init.